### PR TITLE
Feature: beginning

### PR DESCRIPTION
**Pull request by aaryswastaken**
The following modifications have been applied:
 - Architecture expansion
 - Packet management (the very beginning)
 - Implementation of protocol 0x0001 featuring some sort of error management 

To-do before merge:
 - Finish the error handling
 - Tidy for client disconnection
 - Update `PROTOCOL.md`

### CODE REVIEW

**Comment by aaryswastaken** [comment]
+ Setting up CI/CD on teamcity

**Comment by aaryswastaken** [comment]
CI/CD working, fixed error handling

**Comment by aaryswastaken** [comment]
Muted the issues on Teamcity,
> Will be used in later commits

File: PROTOCOL.md [hunk 1]
```diff
@@ -74,3 +74,20 @@ A client can ask for the closing of a room:
...
+
+
+# Error codes
+
+| Code | Description |
+|----|----|
+|`0xa1`| Wrong packet length (handshake) |
+|`0xa2`| Wrong packet identifier (handshake) |
+|`0xa4`| Unsupported version |
+|`0xaa`| Error while receiving the key (minimal length not satisfied |
```
Comment (Portevent) [inline PROTOCOL.md]: )

**Comment by aaryswastaken** [comment]
`panic: runtime error: index out of range [0] with length 0`

lol

**Comment by aaryswastaken** [comment]
Note: there is an issue concerning my system clock. It is now fixed but the commit history is a bit broken.
It broke the linter too, I manually patched it

**Review by Portevent — APPROVED** [review]

**Review by Portevent — APPROVED** [review]

**Pull request merged**